backend/src/process.py

```diff
@@ -197,17 +221,7 @@ async def run_while(
                     raise
                 errors.append(str(e))
 
-        if index < length_estimate:
-            await self.executor.queue.put(
-                {
-                    "event": "iterator-progress-update",
-                    "data": {
-                        "percent": 1,
-                        "iteratorId": self.iterator_id,
-                        "running": None,
-                    },
-                }
-            )
+        await self.__finish_progress(max(length_estimate, index + 1))
```

Comment:
This has to be just `index`.

```suggestion
        await self.__finish_progress(index)
```